cmake/everest-generate.cmake

```diff
@@ -125,6 +119,20 @@ function(ev_add_project)
     )
 endfunction()
 
+macro(ev_add_project)
+    ev_setup_cmake_variables_python_wheel()
+    option(${PROJECT_NAME}_INSTALL_EV_CLI_IN_PYTHON_VENV "Install ev-cli in python venv instead of using system" ON)
+    set(${PROJECT_NAME}_PYTHON_VENV_PATH "${CMAKE_BINARY_DIR}/venv" CACHE PATH "Path to python venv")
+
+    ev_setup_python_executable(
+        USE_PYTHON_VENV ${${PROJECT_NAME}_USE_PYTHON_VENV}
+        PYTHON_VENV_PATH ${${PROJECT_NAME}_PYTHON_VENV_PATH}
+    )
+    setup_ev_cli()
+
```

Comment:
In case of actual defining those vars for example `some-out-of-tree-project_PYTHON_VENV_PATH=${CMAKE_BINARY_DIR}/venv1` and `everest-core_PYTHON_VENV_PATH=${CMAKE_BINARY_DIR}/venv2`, there would be two python virtualvenvs be installed which could lead to undefined behaviour during the build process. Same for `${PROJECT_NAME}_INSTALL_EV_CLI_IN_PYTHON_VENV`.

Maybe we should change to non-project-prefixed vars with global validity as for example:
`EVEREST_INSTALL_EV_CLI_IN_PYTHON_VENV`

Those vars could be setup in a dedicated function which can be triggered from multiple places. This function could be guarded to be only executed once.

Reply:
wasn't this the original intention to be able to specify multiple venvs? Otherwise the ${PROJECT_NAME}_ prefixes would be a bit redundant. For now I can at least report it works well with the in tree and out-of-tree repos, as well as yocto. And doesn't need any more modifications on out-of-tree repos :partying_face: